wire_test.go

```diff
@@ -206,6 +207,86 @@ func TestServerWritingResult(t *testing.T) {
 	})
 }
 
+func TestServerHandlingMultipleConnections(t *testing.T) {
+	address := TConstructMockPsqlServer(t)
+	databaseUrl := fmt.Sprintf("postgres://%s:%d", address.IP, address.Port)
+	conn, err := sql.Open("pgx", databaseUrl)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_ = conn.Close()
+	})
+
+	t.Run("test connectivity", func(t *testing.T) {
+		err := conn.Ping()
+		require.NoError(t, err)
+	})
+
+	t.Run("simple query execution should return rows", func(t *testing.T) {
+		rows, err := conn.Query("select age from person")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			rows.Close()
+		})
+		require.True(t, rows.Next())
+		require.NoError(t, rows.Err())
+	})
+
+	t.Run("prepared statement query execution should support all parameter styles", func(t *testing.T) {
```

Comment:
Small suggestion for this subtest. I would like to suggest to rename this query to prepared statements and extend the test case to include queries without parameters. This would allow us to test the prepared statements cache implementation.

```suggestion
	t.Run("prepared statements", func(t *testing.T) {
```

Reply:
done